### PR TITLE
Add support for timeline bar marks.

### PR DESCRIPTION
**Pull request by jordieh**
This PR implements the timeline bar marks mentioned in #16. I added the following macros:
- \tl@tlcvbar (Creates the timeline bar, exists to avoid code repititon.)
- \tlenablemarksyr
- \tldisablemarksyr
- \tlenablemarksmo
- \tlenablemarksyr
- \tlmarkheightmo (1 parameter, default = 0.2ex)
- \tlmarkheightyr (1 parameter, default = 0.4ex)

By default all marks are disabled and the bar will look like normal.

**Example output**
```LaTeX
\documentclass[11pt,a4paper,sans]{moderncv}
\moderncvstyle{casual}
\moderncvcolor{blue}
\usepackage{moderntimeline}
\tlmaxdates{2017}{2023}
\tlwidth{0.8ex}
\tltext{\tiny}
\name{Test}{Test}
\tlenablemonths
\tlenablemarksyr
\begin{document}
    \tlcventry[blue]{2019}{0}{}{}{}{}{}
    \tlcventry[purple]{2018/3}{2019}{}{}{}{}{}
    \tldatecventry[green]{2019/7}{}{}{}{}{}
    \tlcventry[blue]{2018/1}{2019}{}{}{}{}{}
    \tlcventry[red]{2018/3}{2018/12}{}{}{}{}{}
    \tlcventry[purple]{2018}{2019/12}{}{}{}{}{}
    \tlcventry[green]{2018/12}{2019/1}{}{}{}{}{}
    \tlcventry[green]{2019/1}{2019/2}{}{}{}{}{}
    \tlcventry[red]{2019/1}{2019/12}{}{}{}{}{}
    \tlcventry[purple]{2018}{2019}{}{}{}{}{}
    \tldatecventry[green]{2019}{}{}{}{}{}
    \tldatecventry[blue]{2019/12}{}{}{}{}{}
    \tldatecventry[red]{2019/1}{}{}{}{}{}
\end{document}
```
Example timeline with marks

### CODE REVIEW

**Comment by jordieh** [comment]
@raphink :)

**Comment by raphink** [comment]
Thanks for this PR @jordieh!

Backwards compatibility still works, which is good.

I noticed that your macros require to use `\tlmaxdates` otherwise `\tltickdstyr` is not defined. This is a change in behavior: the current code doesn't need a specific call to `\tlmaxdates` but can instead get these dates from the `firstyear` and `lastyear` parameters in the package call:

```tex
\usepackage[firstyear=1999,lastyear=2020]{moderntimeline}
```

I'm also not sure why you need to redefine `\tl@totalyears`, which is the same as `\tl@yearrange` from what I can tell.

In short, I think you could just do without `\tl@totalyears` (or rename `\tl@yearrange`, as you wish) and compute `\tl@tickdistmo` and `\tl@tickdistyr` outside of `\tlmaxdates` so it gets calculated every time.

File: moderntimeline.dtx [hunk 1]
```diff
@@ -487,6 +558,10 @@
    \def\tl@firstyear{#1}
    \def\tl@lastyear{#2}
    \pgfmathsetmacro\tl@yearrange{\tl@lastyear-\tl@firstyear}
+   \pgfmathsetmacro\tl@totalyears{\tl@lastyear-\tl@firstyear}%
+   \pgfmathsetmacro\tl@totalmonths{12*\tl@totalyears}%
+   \pgfmathsetmacro\tl@tickdistmo{\hintscolumnwidth/\tl@totalmonths}%
+   \pgfmathsetmacro\tl@tickdistyr{\hintscolumnwidth/\tl@totalyears}%
```
Comment (raphink) [inline moderntimeline.dtx:564]: These two lines need to be computed outside of the macro definition, in order to make it optional.

File: moderntimeline.dtx [hunk 2]
```diff
@@ -487,6 +558,10 @@
    \def\tl@firstyear{#1}
    \def\tl@lastyear{#2}
    \pgfmathsetmacro\tl@yearrange{\tl@lastyear-\tl@firstyear}
+   \pgfmathsetmacro\tl@totalyears{\tl@lastyear-\tl@firstyear}%
```
Comment (raphink) [inline moderntimeline.dtx:561]: As stated in my earlier comment, this is a verbatim copy of `\tl@yearrange` above.

File: moderntimeline.dtx [hunk 3]
```diff
@@ -228,6 +228,21 @@
 % the beginning of the year, so is not advisable to use months in dates.
 % Default.
 %
+% \DescribeMacro{\tlenablemarksmo}
+% This macro places vertical lines on the timeline bar at every month and year.
+% You can set the height of the marks with |\tlmarkheightmo| (default = 0.2ex).
+%
+% \DescribeMacro{\tlenablemakrsyr}
```
Comment (raphink) [inline moderntimeline.dtx:235]: Same here, if you can provide a package option, that would be great.

File: moderntimeline.dtx [hunk 4]
```diff
@@ -228,6 +228,21 @@
 % the beginning of the year, so is not advisable to use months in dates.
 % Default.
 %
+% \DescribeMacro{\tlenablemarksmo}
```
Comment (raphink) [inline moderntimeline.dtx:231]: Could you add package options (as a boolean) for this as well?

See l. 357 and 358 for examples of how to declare them.

**Comment by raphink** [comment]
@jordieh Once you've done the changes, would you please add a mention of https://tex.stackexchange.com/a/545111/951 similar to what I've done in the docs before?

The final step will be to adapt the checksum on line 46. It requires to run `pdflatex moderntimeline.dtx` and copy the checksum that is provided by the output.

**Comment by jordieh** [comment]
> @jordieh Once you've done the changes, would you please add a mention of https://tex.stackexchange.com/a/545111/951 similar to what I've done in the docs before?
I am sorry, what do you mean by this exactly?

I will probably implement the requested changes tomorrow.

**Comment by raphink** [comment]
@jordieh sorry my bad, I mixed up the requests for changes on this lib! Ignore my comment!

**Comment by jordieh** [comment]
> @jordieh sorry my bad, I mixed up the requests for changes on this lib! Ignore my comment!

No problem, should i increment the version and edit the changelog?

**Comment by jordieh** [comment]
Two weeks later...

I added the package options alongside the other macros:
```latex
\usepackage[marksmo=1.6ex]{moderntimeline} % (years appear to be disabled), moheight=1.6ex
\usepackage[marksyr=1.6ex, marksmo=2.0ex]{moderntimeline} % yrheight=1.6ex, moheight=2.0ex
\usepackage[marksyr, marksmo]{moderntimeline} % yrheight=0.4ex, moheight=0.2ex
\usepackage[marksyr]{moderntimeline} % yrheight=0.4ex, months disabled
```
There are two things that still bother me;
1. When 'disabling' years and 'enabling' months, the month marks appear on the position of the year marks because of the implementation, should this be fixed? I don't think it's something that will bother a lot of people.
2. When enabling year marks and having a fading end, this happens:

**Comment by jordieh** [comment]
@raphink Just pinging you because it has been two weeks, if you agree with the changes, i will add a bit to the fading bar to override the bar mark and edit the checksum.

**Comment by raphink** [comment]
Right, the `+1` is kind of a hack but it does the trick nicely I think.

File: moderntimeline.dtx [hunk 5]
```diff
@@ -102,12 +102,14 @@
 % and can be loaded with:
 %
 % \begin{verbatim}
-%    \usepackage[firstyear=1999,lastyear=2012]{moderntimeline}
+%    \usepackage[firstyear=1999,lastyear=2012,marksyr=0.4ex,marksmo=0.2ex]{moderntimeline}
```
[the remark this answers is not in the document]
Comment (jordieh) [reply moderntimeline.dtx]: Is the trailing comma on the 2th line supposed to be there?

**Comment by raphink** [comment]
Thanks for this and for your patience in preparing the feature!

I've prepared a v0.11 release.

**Comment by jordieh** [comment]
No problem, issue #16 should probably be closed though :D. Thanks for maintaning this package.
